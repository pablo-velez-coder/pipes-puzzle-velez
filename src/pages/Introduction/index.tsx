import { useState } from 'react'
import { useDispatch } from 'react-redux'
import { types } from 'src/types'
import { formattingName } from 'src/utils/arrangingMap'
import styles from './styles.module.scss'

const Introduction = () => {

    const [name, setName] = useState('')
    const dispatch = useDispatch()

    const handleSubmit = (e : React.SyntheticEvent) =>{
        e.preventDefault();
        dispatch({
            type: types.SAVE_USER,
            payload: formattingName(name)
        })
    }

    return (
        <div className={styles.introductionPage}>
          <div className={styles.introductionPageContainer}>
          <h1 className="title">Pipes Puzzle</h1>
            <p className="input-name">To begin, tell us your name!</p>
            <form
            onSubmit={handleSubmit}
            >
                <input 
                placeholder='name'
                name="name"
                value={name}
                onChange={e=> setName(e.target.value)}
                />
                <button 
                data-id="submit-button"
                disabled={name.length === 0}
                type='submit'>
                    Go!
                </button>
            </form>
          </div>
        </div>
    )
}

export default Introduction
