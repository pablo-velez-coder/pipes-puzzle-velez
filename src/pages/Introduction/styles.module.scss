@import "src/scss/mixins.scss";
@import "src/style/colors.scss";

.introductionPage{
    background-color: $main-background-color;
    height:100vh;
    display:grid;
    place-items: center;
    &Container{
        background-color: $main-background-container;
        border-radius: 5px;
        padding:1.5rem;
        text-align: center;
        >h1{
            font-size: 5rem;
            margin-bottom: 3rem;
        }
        p{
            color:#fff;
            margin-bottom: 1.5rem;
            font-style: italic;
        }
        form{
            >input{
                @include cartoonish-style;
                padding:6px 22px;
                border-radius: 4px;
                outline-width: 0;
                margin-right: 6px;
            }
            >button{
                @include cartoonish-style;
                border-radius: 50%;
                padding:1rem;
                cursor: pointer;
                font-weight: bold;
                font-size: 1.1rem;
  
            }
        }
    }
}