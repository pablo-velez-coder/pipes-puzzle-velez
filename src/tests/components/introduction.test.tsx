import { shallow, mount } from "enzyme";
import { Provider } from "react-redux";
import Introduction from "src/pages/Introduction";
import store from "src/redux/store";

describe('test in introduction component',()=>{

    const wrapper = shallow(
        <Provider store={store}>
        <Introduction />
        </Provider>
).dive()

const introPage =  mount(
    <Provider store={store}>
    <Introduction />
    </Provider>)

    test('should render correctly', () => {
        expect(wrapper).toMatchSnapshot()
    });

    test('should display texts in the dom', () => {
      const gameTitle = introPage.find('.title').text()
      const callToInput = introPage.find('.input-name').text()
      const input = introPage.find("input[name='name']")
      const button = introPage.find("button[data-id='submit-button']")
        expect(gameTitle).toBe('Pipes Puzzle')
        expect(callToInput).toBe('To begin, tell us your name!')
        expect(input.exists()).toBe(true)
        expect(button.text()).toBe('Go!')
    });

    test('should simulate input a name and succesful submitting', () => {
        const name = 'Pablo'
        const input = introPage.find("input[name='name']")
        const button = introPage.find("button[data-id='submit-button']")
        input.simulate('change',{target:{value: name}})
        button.simulate('submit')

    });
    
    
    
})