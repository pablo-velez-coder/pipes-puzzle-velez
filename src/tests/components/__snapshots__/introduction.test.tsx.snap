// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in introduction component should render correctly 1`] = `ShallowWrapper {}`;
